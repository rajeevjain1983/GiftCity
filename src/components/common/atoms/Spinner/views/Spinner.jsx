// 9fbef606107a605d69c0edbcd8029e5d 
import React from 'react';
import PropTypes from 'prop-types';
import withStyles from '../../../hoc/withStyles';
import styles from '../styles/Spinner.style';

const Spinner = ({ className }) => {
  return (
    <div className={className}>
      <div className="tcp-circle">
        <div />
        <div />
        <div />
        <div />
      </div>
    </div>
  );
};

Spinner.propTypes = {
  className: PropTypes.string,
};
Spinner.defaultProps = {
  className: '',
};

export default withStyles(Spinner, styles);
