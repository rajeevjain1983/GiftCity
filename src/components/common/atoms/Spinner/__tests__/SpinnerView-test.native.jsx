// 9fbef606107a605d69c0edbcd8029e5d 
import React from 'react';
import { Animated } from 'react-native';
import { shallow } from 'enzyme';
import { SpinnerViewVanilla } from '../views/Spinner.native';

describe('SpinnerView Native Componenet', () => {
  let component;

  beforeEach(() => {
    component = shallow(<SpinnerViewVanilla />);
  });

  it('should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('should return Animated component value four', () => {
    expect(component.find(Animated)).toHaveLength(0);
  });

  it('should return Styled(View) component value one', () => {
    expect(component.find('Styled(View)')).toHaveLength(1);
  });
});
