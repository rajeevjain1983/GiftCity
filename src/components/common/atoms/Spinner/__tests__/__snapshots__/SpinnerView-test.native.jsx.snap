// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpinnerView Native Componenet should match snapshot 1`] = `
<Styled(View)>
  <ForwardRef(AnimatedComponentWrapper)
    style={
      Object {
        "height": 0,
        "width": 0,
      }
    }
  />
  <ForwardRef(AnimatedComponentWrapper)
    style={
      Array [
        Object {
          "height": 8,
          "width": 8,
        },
        Object {
          "left": 0,
          "top": 0,
        },
      ]
    }
  />
  <ForwardRef(AnimatedComponentWrapper)
    style={
      Array [
        Object {
          "height": 8,
          "marginLeft": 10,
          "width": 8,
        },
        Object {
          "left": 0,
          "top": 0,
        },
      ]
    }
  />
  <ForwardRef(AnimatedComponentWrapper)
    style={
      Array [
        Object {
          "height": 8,
          "marginLeft": 17,
          "width": 8,
        },
      ]
    }
  />
</Styled(View)>
`;
